File: src/devng/projecteuler/problem001/problem001_test.go
package main

import "testing"

func TestSumN(t *testing.T) {
	sum10 := SumN(10)
	if sum10 != 55 {
		t.Error("The sum of the first 10 numbers must be 55.")
	}

	sum1000 := SumN(1000)
	if sum1000 != 500500 {
		t.Error("The sum of the first 10 numbers must be 500500.")
	}
}

func TestSum35(t *testing.T) {
	sum10 := Sum35(10)
	if sum10 != 23 {
		t.Error("The sum of multiples of 3 and 5 for the first 10 numbers must be 23.")
	}

	sum1000 := Sum35(1000)
	if sum1000 != 233168 {
		t.Error("The sum of multiples of 3 and 5 for the first 10 numbers must be 233168.")
	}
}

File: src/devng/projecteuler/problem001/problem001_test.py
from problem001 import *
import unittest

class Problem001Test(unittest.TestCase):
    def test_sum_n(self):
        sum10 = sum_n(10)
        self.assertEqual(sum10, 55)

    	sum1000 = sum_n(1000)
        self.assertEqual(sum1000, 500500)

    def test_sum35(self):
        sum10 = sum_3_5(10)
        self.assertEqual(sum10, 23)

    	sum1000 = sum_3_5(1000)
        self.assertEqual(sum1000, 233168)

if __name__ == '__main__':
    unittest.main()
